File: backend/dataset/reverse/reverse_8.py
class Solution:
    def reverse(self, x: int) -> int:
        output = ''
        str_x = str(x)
        if str_x[0] == '-':
            output+='-'
            for i in range(len(str_x)-1,0,-1):
                output+=str_x[i]
        else:
            for j in range(len(str_x)-1,-1,-1):
                output+=str_x[j]
        if int(output) not in range(-2147483648,2147483648):
            return 0
        return int(output)

File: backend/CDataset/reverse/reverse_8.cpp
class Solution {
public:
    int reverse(int x) {
        if(x==0)return 0;
        long long int n=x;
        if(n<0)n=-n;
        long long int ans=0;
        while(n>0){
            ans=ans*10+n%10;
            n=n/10;
            if(ans>pow(2,31)-1)return 0;
        }
        if(x<0)return -ans;
        return ans;
    }
};